apps/election-manager/src/utils/semsTallies.ts

```diff
@@ -333,3 +340,53 @@ export function getPrecinctIdsInExternalTally(
   const resultsByPrecinct = tally.resultsByCategory.get(TallyCategory.Precinct)
   return resultsByPrecinct ? Object.keys(resultsByPrecinct) : []
 }
+
+export function filterExternalTalliesByParams(
+  fullTally: OptionalFullElectionExternalTally,
+  election: Election,
+  {
+    precinctId,
+    partyId,
+    scannerId,
+  }: { precinctId?: string; partyId?: string; scannerId?: string }
+): OptionalExternalTally {
+  if (!fullTally || scannerId) {
+    return undefined
+  }
+
+  const { overallTally, resultsByCategory } = fullTally
+
+  let filteredTally = overallTally
+
+  if (precinctId) {
+    filteredTally =
+      resultsByCategory.get(TallyCategory.Precinct)?.[precinctId] ||
+      getEmptyTally()
+  }
+
+  if (!partyId) {
+    return filteredTally
+  }
+
+  // Filter contests by party and recompute the number of ballots based on those contests.
+  const districtsForParty = getDistrictIdsForPartyId(election, partyId)
+  const filteredContestTallies: Dictionary<ContestTally> = {}
+  Object.keys(filteredTally.contestTallies).forEach((contestId) => {
+    const contestTally = filteredTally.contestTallies[contestId]
+    if (
+      !!contestTally &&
+      districtsForParty.includes(contestTally?.contest.districtId) &&
+      contestTally?.contest.partyId === partyId
```

Comment:
```suggestion
      contestTally &&
      districtsForParty.includes(contestTally.contest.districtId) &&
      contestTally.contest.partyId === partyId
```

Unless you need a `boolean`, using `!!` is superfluous. Also, TS can infer that `contestTally` is not `undefined` which makes the optional chaining unnecessary (and confusing).